### ISSUE TITLE
Surprising inconsistency between disk_partitions() and disk_io_counters(True) on OSX

### ISSUE THREAD

**Issue by gar1t** [giampaolo/psutil#911]
I'm seeing this on Darwin mac 15.5.0 Darwin Kernel Version 15.5.0 with psutil v4.3.1.

```
>>> psutil.disk_partitions()
[sdiskpart(device='/dev/disk1', mountpoint='/', fstype='hfs',
           opts='rw,local,rootfs,dovolfs,journaled,multilabel')]

>>> psutil.disk_io_counters(True)
{'disk0': sdiskio(...),
 'disk2': sdiskio(...),
 'disk4': sdiskio(...)}
```

**Comment by giampaolo** [comment]
Try to `psutil.disk_partitions(all=True)`.

**Comment by warsaw** [comment]
@giampaolo I would like to reopen this issue.  With psutil 5.3.1 there seems to be no way to map between the device names given by `disk_partitions(True)` and `disk_io_counters(True)`:

```
>>> list(disk_io_counters(perdisk=True))
['disk0']
>>> list(disk.device for disk in disk_partitions(all=True))
['/dev/disk1', 'devfs', 'map -hosts', 'map auto_home', 'map -fstab']
```

This is on macOS 10.12.

Here's a use case.  Let's say I want to calculate disk utilization on the disk that contains a particular directory.  How can I do that?  `disk_io_counters(True)` can give me metrics on all the disks, but its keyed off of a name that I can't use to map back to a particular file.  I can get the `st_dev` of a file using `os.stat()`, and I can match that against the `st_dev`s of a `disk.mountpoint` that comes back from `disk_partitions(True)`.

But once I've found the `sdiskpart` for the device containing the file, how to I map that back to the information returned by `disk_io_counters(True)`?  How do I know definitively what the `disk0` key in that output is referring to?

**Comment by giampaolo** [comment]
Makes sense to me. Won't have time to look into this any time soon though.

**Comment by davidnewhall** [comment]
Same problem in the go library version on macOS: https://github.com/shirou/gopsutil/issues/855
